src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -211,7 +212,25 @@ impl fmt::Display for LocationState {
         Ok(())
     }
 }
-
+/// state associated with each location of the allocation
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Location {
+    /// Maps a tag and a location to a perm, with possible lazy
+    /// initialization.
+    ///
+    /// NOTE: not all tags registered in `nodes` are necessarily in all
+    /// ranges of `rperms`, because `rperms` is in part lazily initialized.
+    /// Just because `nodes.get(key)` is `Some(_)` does not mean you can safely
+    /// `unwrap` any `perm.get(key)`.
+    ///
+    /// We do uphold the fact that `keys(perms)` is a subset of `keys(nodes)`
+    pub perms: UniValMap<LocationState>,
+    /// Maps a tag and a location to its wildcard access tracking information,
```

Comment:
This comment is wrong -- this maps a tag to its wildcard access info, but the location mapping is handled elsewhere. Same above.